mixer/adapter/redisquota/redisquota.go

```diff
@@ -0,0 +1,400 @@
+		option.PoolSize = int(b.adapterConfig.ConnectionPoolSize)
+	}
+
+	client := redis.NewClient(&option)
+	if _, err := client.Ping().Result(); err != nil {
+		ce = ce.Appendf(info.Name, "could not create a connection to redis server: %v", err)
+		return
+	}
+
+	// check scripts loading to redis
+	scripts := make(map[config.Params_QuotaAlgorithm]*redis.Script, 2)
+	for algorithm, script := range rateLimitingLUAScripts {
+		scripts[algorithm] = redis.NewScript(script)
+		if _, err := scripts[algorithm].Load(client).Result(); err != nil {
+			ce = ce.Appendf(info.Name, "unable to initialized redis service: %v", err)
+			return
+		}
+	}
+
+	_ = client.Close()
+	return
+}
+
+// getOverrideHash returns hash key of the given dimension in sorted by key
+func getDimensionHash(dimensions map[string]string) (string, error) {
+	var keys []string
+	for k := range dimensions {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+
+	h := fnv.New32a()
+	for _, key := range keys {
+		if _, err := io.WriteString(h, key+"\t"+dimensions[key]+"\n"); err != nil {
+			return "", err
+		}
+	}
+	return strconv.Itoa(int(h.Sum32())), nil
+}
+
+func (b *builder) Build(context context.Context, env adapter.Env) (adapter.Handler, error) {
+	limits := make(map[string]*config.Params_Quota, len(b.adapterConfig.Quotas))
+	for idx := range b.adapterConfig.Quotas {
+		limits[b.adapterConfig.Quotas[idx].Name] = &b.adapterConfig.Quotas[idx]
+	}
+
+	// Build memory address of dimensions to hash map
+	dimensionHash := make(map[*map[string]string]string)
+	for key := range limits {
+		for index := range limits[key].Overrides {
+			if hash, err := getDimensionHash(limits[key].Overrides[index].Dimensions); err == nil {
+				dimensionHash[&(limits[key].Overrides[index].Dimensions)] = hash
+			}
+		}
+	}
+
+	// initialize redis client
+	option := redis.Options{
+		Addr: b.adapterConfig.RedisServerUrl,
+	}
+
+	if b.adapterConfig.ConnectionPoolSize > 0 {
+		option.PoolSize = int(b.adapterConfig.ConnectionPoolSize)
+	}
+
+	client := redis.NewClient(&option)
+
+	// load scripts into redis
+	scripts := make(map[config.Params_QuotaAlgorithm]*redis.Script, 2)
+	for algorithm, script := range rateLimitingLUAScripts {
+		scripts[algorithm] = redis.NewScript(script)
+	}
+
+	h := &handler{
+		client:        client,
+		limits:        limits,
+		scripts:       scripts,
+		logger:        env.Logger(),
+		getTime:       time.Now,
+		dimensionHash: dimensionHash,
+	}
+
+	return h, nil
+}
+
+////////////////// Runtime Methods //////////////////////////
+
+// matchDimensions matches configured dimensions with dimensions of the instance.
+func matchDimensions(cfg *map[string]string, inst *map[string]interface{}) bool {
+	for k, val := range *cfg {
+		rval := (*inst)[k]
+		if rval == val { // this dimension matches, on to next comparison.
+			continue
+		}
+
+		// if rval has a string representation then compare it with val
+		// For example net.ip has a useful string representation.
+		switch v := rval.(type) {
```

Comment:
this would panic in case the override dimensions has label that is not part of the instance dimensions. We must validate this during the validation code.

Reply:
Added that checks the existence of rval